cdap-cli/src/main/java/co/cask/cdap/cli/CLIConfig.java

```diff
@@ -155,14 +171,19 @@ private AccessToken acquireAccessToken(ClientConfig clientConfig, ConnectionConf
     }
 
     try {
-      AccessToken savedAccessToken = getSavedAccessToken(connectionInfo.getHostname());
-      checkConnection(clientConfig, connectionInfo, savedAccessToken);
-      return savedAccessToken;
+      UserAccessToken savedToken = getSavedAccessToken(connectionInfo.getHostname());
+
+      checkConnection(clientConfig, connectionInfo, savedToken.getAccessToken());
+      connectionInfo.setUsername(savedToken.getUsername());
```

Comment:
I think the user name should already been set when the `ConnectionInfo` object was instantiated. See my comment below.